### ISSUE TITLE
Why is the batch normalization branch code slow ?

### ISSUE THREAD

**Issue by pdcoded** [mozilla/DeepSpeech#550]
It supposed to be faster than dropout configuration by theory. But it seemed to be fairly slower on the ldc93s1 data.

**Comment by kdavis-mozilla** [comment]
I think you're comparing the ldc93s1 test on master to the DS2 code. The ldc93s1 is a much smaller network

**Comment by pdcoded** [comment]
No no @kdavis-mozilla  . I am comparing bnlstm branch to master on the same ldc93s1. And the training time is much slower in bnlstm. What's the reason? Any specific performance comparison you have ?

**Comment by kdavis-mozilla** [comment]
As I mentioned before ldc93s1 is a much smaller network. The ldc93s1 network has a width of 494[[1](https://github.com/mozilla/DeepSpeech/blob/master/bin/run-ldc93s1.sh#L22)]. The bnlstm network has a width of 2048[[2](https://github.com/mozilla/DeepSpeech/blob/bnlstm/DeepSpeech.py#L186)].

**Comment by pdcoded** [comment]
Didn't happen to check that :) Any performance comparison of both that has been done?

**Comment by kdavis-mozilla** [comment]
When both networks made about the same size the DS2 architecture is faster. We haven't done a detail analysis as the DS2 architecture is still under development.

**Comment by pdcoded** [comment]
Also any update onto when can expect first stable version of DS2 ?

**Comment by kdavis-mozilla** [comment]
@pdcoded discussion like this should be handled on the IRCH channel

**Comment by pdcoded** [comment]
sure . Doing that
